src/main/java/eu/hbp/mip/controllers/PathologiesAPI.java

```
@@ -35,41 +33,90 @@ public class PathologiesAPI {
+        for (String pathology : mipEnginePathologyAttributes.keySet()) {
+            pathologyDTOS.add(new PathologyDTO(pathology, mipEnginePathologyAttributes.get(pathology), datasetsPerPathology.get(pathology)));
         }
 
         // If authentication is disabled return everything
         if (!authenticationIsEnabled) {
-            logger.LogUserAction("Successfully loaded " + allPathologies.size() + " pathologies");
-            return ResponseEntity.ok().body(gson.toJson(allPathologies));
+            logger.LogUserAction("Successfully loaded " + pathologyDTOS.size() + " pathologies");
+            return ResponseEntity.ok().body(gson.toJson(pathologyDTOS));
         }
 
         logger.LogUserAction("Successfully loaded all authorized pathologies");
-        return ResponseEntity.ok().body(ClaimUtils.getAuthorizedPathologies(logger,  authentication, allPathologies));
+        return ResponseEntity.ok().body(ClaimUtils.getAuthorizedPathologies(logger, authentication, pathologyDTOS));
+    }
+
+    public Map<String, List<PathologyDTO.PathologyDatasetDTO>> getMIPEngineDatasetsPerPathology(Logger logger) {
+        Map<String, Map<String, MetadataHierarchyDTO.CommonDataElement>> mipEngineCDEsMetadata;
+        // Get MIPEngine algorithms
+        try {
+            StringBuilder response = new StringBuilder();
+            HTTPUtil.sendGet(mipengineCDEsMetadataUrl, response);
+            mipEngineCDEsMetadata = gson.fromJson(
+                    response.toString(),
+                    new TypeToken<HashMap<String, Map<String, MetadataHierarchyDTO.CommonDataElement>>>() {
+                    }.getType()
+            );
+        } catch (ConnectException e) {
+            logger.LogUserAction("An exception occurred: " + e.getMessage());
+            return null;
+        } catch (IOException e) {
+            logger.LogUserAction("An exception occurred: " + e.getMessage());
+            return null;
+        }
```

Catch all exceptions here, logging the exception and returning null.